ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/DeprecatedStatementsCheckerTest.java

```diff
@@ -0,0 +1,357 @@
+        STREAM_1.text(),
+        STREAM_2.text(),
+        STREAM_3.text()
+    ));
+
+    // When
+    final Optional<DeprecatedStatementsChecker.Deprecations> deprecations =
+        statementsChecker.checkStatement(statement);
+
+    // Then
+    assertThat(deprecations, is(Optional.of(DEPRECATED_STREAM_STREAM_OUTER_JOIN_WITH_NO_GRACE)));
+  }
+
+  @Test
+  public void shouldDeprecateStreamStreamLeftJoinWithNoGraceOnMultiStreamStreamTableJoin() {
+    // Given
+    final Statement statement = parse(String.format(
+        "SELECT * FROM %s AS l "
+            + "LEFT JOIN %s AS m WITHIN 1 SECOND ON l.K = m.K "
+            + "LEFT JOIN %s AS r ON l.K = r.K;",
+        STREAM_1.text(),
+        STREAM_2.text(),
+        TABLE_1.text()
+    ));
+
+    // When
+    final Optional<DeprecatedStatementsChecker.Deprecations> deprecations =
+        statementsChecker.checkStatement(statement);
+
+    // Then
+    assertThat(deprecations, is(Optional.of(DEPRECATED_STREAM_STREAM_OUTER_JOIN_WITH_NO_GRACE)));
+  }
+
+  @Test
+  public void shouldNotDeprecateStreamStreamLeftJoinWithGrace() {
+    // Given
+    final Statement statement = parse(String.format(
+        "SELECT * FROM %s AS l "
+            + "LEFT JOIN %s AS r WITHIN 1 SECOND GRACE PERIOD 1 SECOND ON l.K = r.K;",
+        STREAM_1.text(),
+        STREAM_2.text()
+    ));
+
+    // When
+    final Optional<DeprecatedStatementsChecker.Deprecations> deprecations =
+        statementsChecker.checkStatement(statement);
+
+    // Then
+    assertThat(deprecations, is(Optional.empty()));
+  }
+
+  @Test
+  public void shouldNotDeprecateStreamStreamLeftJoinWithGraceOnMultiStreamStreamJoin() {
+    // Given
+    final Statement statement = parse(String.format(
+        "SELECT * FROM %s AS l "
+            + "LEFT JOIN %s AS m WITHIN 1 SECOND GRACE PERIOD 1 SECOND ON l.K = m.K "
+            + "LEFT JOIN %s AS r WITHIN 1 SECOND GRACE PERIOD 1 SECOND ON m.K = r.K;",
+        STREAM_1.text(),
+        STREAM_2.text(),
+        STREAM_3.text()
+    ));
+
+    // When
+    final Optional<DeprecatedStatementsChecker.Deprecations> deprecations =
+        statementsChecker.checkStatement(statement);
+
+    // Then
+    assertThat(deprecations, is(Optional.empty()));
+  }
+
+  @Test
+  public void shouldNotDeprecateStreamStreamLeftJoinWithGraceOnMultiStreamStreamTableJoin() {
+    // Given
+    final Statement statement = parse(String.format(
+        "SELECT * FROM %s AS l "
+            + "LEFT JOIN %s AS m WITHIN 1 SECOND GRACE PERIOD 1 SECOND ON l.K = m.K "
+            + "LEFT JOIN %s AS r ON l.K = r.K;",
+        STREAM_1.text(),
+        STREAM_2.text(),
+        TABLE_1.text()
+    ));
+
+    // When
+    final Optional<DeprecatedStatementsChecker.Deprecations> deprecations =
+        statementsChecker.checkStatement(statement);
+
+    // Then
+    assertThat(deprecations, is(Optional.empty()));
+  }
+
+  @Test
+  public void shouldDeprecateStreamStreamOuterJoinWithoutGrace() {
```

Comment:
If we treat all three joins (inner/left/outer) equal as proposed about, those tests seems to be redundant?